lib/elixir/lib/module/types/pattern.ex

```diff
@@ -380,6 +525,19 @@ defmodule Module.Types.Pattern do
     Map.fetch!(@guard_functions, {name, arity})
   end
 
+  defp filter_clauses(signature, expected, stack, context) do
+    clauses =
+      Enum.filter(signature, fn {_params, return} ->
+        match?({:ok, _type, _context}, unify(return, expected, stack, context))
+      end)
+
+    if clauses == [] do
+      signature
```

Comment:
Do you think this is the best way to go? Right now it is not a problem, because if we reach this clause it means we will print a warning (that we hope the user will address) but I am afraid that we will be doing work that we know will fail in the future. I would prefer to return an empty clause and already convert it to an error upstream. Thoughts?

Reply:
I have removed it and everything passes, let's see if we can keep it that way.